com.unity.sg2/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/Input/Matrix/TransformationMatrixNode.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Collections.Generic;
+using UnityEditor.ShaderGraph.GraphDelta;
+
+namespace UnityEditor.ShaderGraph.Defs
+{
+    internal class TransformationMatrixNode : IStandardNode
+    {
+        static string Name = "TransformationMatrix";
+        static int Version = 1;
+        public static FunctionDescriptor FunctionDescriptor => new(
+            Version,
+            Name,
+            "Out = Matrix;",
+            new ParameterDescriptor("Out", TYPE.Mat4, GraphType.Usage.Out),
+            new ParameterDescriptor("Matrix", TYPE.Mat4, GraphType.Usage.Static, defaultValue: REF.M)
```

Comment:
`Shader error in 'Test': cannot implicitly convert from 'float4x4' to 'float3' at line 2057 (on d3d11)` error thrown when popping out the node. Might be due to the default values aren't fully supported yet, just pointing out in case

Reply:
I think this is happening because it's trying to render a preview with a Matrix.  I forgot to add the hasPreview: false flag. Thanks for the reminder!